### DISCUSSION TITLE
Gallery for Agency Theme

### DISCUSSION THREAD

**Discussion by AlexeyGorbachev** [OrchardCMS/OrchardCore#13236] [Q&A]
Hi all! Does anybody implement highslide js (http://highslide.com/) or some other gallery with categories? For example, in Agency theme there is portfolio part of landing page and project data type. But projects are not grouped by categories (for example two images in one group, three in other groups).  I have found [Etch.OrchardCore.Gallery](https://github.com/EtchUK/Etch.OrchardCore.Gallery). But where are examples of usage?

**Comment by DrewBrasher** [comment]
I recently used Etch.OrchardCore.Gallery but I also couldn't find any documentation on setting up the lightgallery front end. There is an issue in their GitHub for adding the documentation here: https://github.com/EtchUK/Etch.OrchardCore.Gallery/issues/14
But like is mentioned in the issue, lightgallery is not free for commercial use so I didn't look too much into setting it up.

I ended up still using that module but just writing a simple front end for it that just uses bootstrap to display the gallery. For that I just had to create a `GalleryPart` template with this liquid code in it:
```
{% assign smCol = 2 %}
{% assign mdCol = 4 %}
{% if Model.MediaItems.size == 1 %}
    {% assign mdCol = 1 %}
    {% assign smCol = 1 %}
{% elsif  Model.MediaItems.size == 2 %}
    {% assign mdCol = 2 %}
{% elsif  Model.MediaItems.size == 3 %}
    {% assign mdCol = 3 %}
{% endif %}

<div class="row row-cols-{{ smCol }} row-cols-md-{{ mdCol }} g-4">
    {% for media in Model.MediaItems %}
        <div class="col text-center">
            {% if media.TypeName == 'Video' %}
                <a href="{{ media.Url }}" target="_blank">
                    {{ media.Title}}
                    <img src="{{ media.Thumb }}" class="img-fluid" />
                </a>
            {% else %}
                <img src="{{ media.Thumb }}" 
                    class="img-fluid" 
                    alt="..." 
                    data-bs-toggle="modal" 
                    data-bs-target="#photoModel" 
                    data-bs-url="{{ media.Url }}"
                    data-bs-type="{{ media.TypeName }}"
                    style="cursor: pointer;">
            {% endif %}
        </div>
    {% endfor %}
</div>

<!-- Modal -->
<div class="modal fade" id="photoModel" tabindex="-1" aria-labelledby="photoModelLabel" aria-hidden="true">
  <div class="modal-dialog modal-lg modal-dialog-centered">
    <div class="modal-content">
      <div class="modal-body">
        <img src="" class="img-fluid">
      </div>
    </div>
  </div>
</div>

{% scriptblock at: "Foot" %}
    var photoModel = document.getElementById('photoModel')
    photoModel.addEventListener('show.bs.modal', function (event) {
    var url = event.relatedTarget.getAttribute('data-bs-url')
    var modalBodyImg = photoModel.querySelector('.modal-body img')
    modalBodyImg.src = url
    })
{% endscriptblock %}
```

You can see how it looks here: https://drewbrasher.com/blog/moss
